### PR TITLE
Remove Job from History API

### PR DESCRIPTION
**Pull request by HuKeping**
a part of issue  #12151 

Signed-off-by: Hu Keping hukeping@huawei.com

### CODE REVIEW

**Comment by cpuguy83** [comment]
Probably some integration test that needs to be modified here to clear up janky.

**Comment by HuKeping** [comment]
yeah, i am working on that :)

File: graph/history.go [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
 	"github.com/docker/docker/image"
 	"github.com/docker/docker/utils"
 )
 
-func (s *TagStore) CmdHistory(job *engine.Job) error {
-	if n := len(job.Args); n != 1 {
-		return fmt.Errorf("Usage: %s IMAGE", job.Name)
+func (s *TagStore) CmdHistory(name string) ([]*types.ImageHistory, error) {
+	if name == "" {
+		return nil, fmt.Errorf("Usage: docker history [OPTIONS] IMAGE")
```
Comment (LK4D4) [inline graph/history.go]: This is sorta weird error. It is about old jobs syntax. I think it is okay to return error right from `s.LookupImage`.

File: graph/history.go [hunk 2]
```diff
@@ -1,24 +1,21 @@
...
 	"github.com/docker/docker/api/types"
-	"github.com/docker/docker/engine"
 	"github.com/docker/docker/image"
 	"github.com/docker/docker/utils"
 )
 
-func (s *TagStore) CmdHistory(job *engine.Job) error {
-	if n := len(job.Args); n != 1 {
-		return fmt.Errorf("Usage: %s IMAGE", job.Name)
+func (s *TagStore) CmdHistory(name string) ([]*types.ImageHistory, error) {
```
Comment (LK4D4) [inline graph/history.go]: Can we rename it to `History`?

Comment (HuKeping) [inline graph/history.go]: OK, `History` is better

**Comment by HuKeping** [comment]
ping@LK4D4 updated

**Comment by duglin** [comment]
LGTM

**Comment by cpuguy83** [comment]
Yes, it looks like that test wasn't really testing anything to begin with.... other than that you can create a job and pass in an arg to it.... definitely good to delete it.

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**